OpenRA.Mods.Common/Traits/Upgrades/DeployToUpgrade.cs

```diff
@@ -89,59 +100,27 @@ public Order IssueOrder(Actor self, IOrderTargeter order, Target target, bool qu
 
 		public void ResolveOrder(Actor self, Order order)
 		{
-			if (order.OrderString != "DeployToUpgrade")
+			if (order.OrderString != "DeployToUpgrade" || deployState == DeployState.Deploying)
 				return;
 
 			if (!IsOnValidTerrain())
```

Comment:
If the order is queued this check doesn't make sense. Check `order.TargetLocation` instead of `self.Location`.